### PR TITLE
Support encrypted values in input configuration

### LINKED ISSUE #14433: Encryption at rest of HTTP headers in config for "JSON path value from HTTP API" input
## What?

Provide an additional field to store sensitive headers in the input configuration for the JSON path input. The value of the field should be stored encrypted in MongoDB.

Because supporting this involves adding a new field type to input configuration, take care to support this in the Graylog Enterprise Forwarder as well.

## Why?

HTTP headers configured for the JSON path input may contain credentials.

### PR DESCRIPTION
**Pull request by thll**
This PR adds generic support of encryption at rest for sensitive input configuration.

A new configuration parameter is added to the HTTP JSON Path input as a first use case. It allows specifying a sensitive HTTP header, e.g. for a bearer token. The field will be stored encrypted, and its content won't be exposed anymore via the REST API once set.

fixes #14433 https://github.com/Graylog2/graylog-plugin-cloud/issues/1086 

Contains upgrade notes for dependent PR Graylog2/graylog-plugin-enterprise#4590

TODO:
- [ ] forwarder support
  https://github.com/Graylog2/graylog-plugin-enterprise/pull/4590
  https://github.com/Graylog2/forwarder/pull/89
- [x] content pack support
- [ ] UI (https://github.com/Graylog2/graylog-plugin-cloud/issues/1086)
---
**!! Don't merge until the UI is completed !!**

----

### CODE REVIEW

**Review by AntonEbel — CHANGES_REQUESTED** [review]

**Review by AntonEbel — APPROVED** [review]
Backend LGTM.

**Comment by grotlue** [comment]
I added the support of encrypted values in the Input Configuration form and view.

https://user-images.githubusercontent.com/3169354/218528210-89930b5f-5ea7-442c-b73e-6196d6f7b50d.mp4

**Review by ousmaneo — CHANGES_REQUESTED** [review]
Looks good! I added some comments

File: graylog2-web-interface/src/components/configurationforms/index.jsx [hunk 1]
```diff
@@ -15,7 +15,7 @@
  * <http://www.mongodb.com/licensing/server-side-public-license>.
  */
 export { default as BooleanField } from './BooleanField';
-export { default as ConfigurationForm } from './ConfigurationForm';
+export { default as ConfigurationForm } from './ConfigurationForm'; // eslint-disable-line import/no-cycle
```
[the remark this answers is not in the document]
Comment (grotlue) [reply graylog2-web-interface/src/components/configurationforms/index.jsx:18]: That is true, but I think it is complaining in the cloud-inputs branch, which is based on this one. I thought I'd change it here already.

**Review by ousmaneo — APPROVED** [review]
Tested it again and I think we good to go 🎉

**Pull request merged**